src/io.c

```diff
@@ -85,3 +83,14 @@ chomp(char *string, ssize_t length)
 		string[length - 1] = '\0';
 	}
 }
+
+static char *
+eager_strpbrk(const char *string, const char *separators) {
+	char *ptr = NULL, *tmp_ptr;
+	for (tmp_ptr = strpbrk(string, separators);
+	     tmp_ptr;
+	     tmp_ptr = strpbrk(tmp_ptr, separators)) {
```

Comment:
Having done far more C++ than C I’m definitely biased towards `for` loops. However, in this particular case I believe that `for` loop actually improves readability. Your suggested implementation is incomplete, since `strpbrk` can return `NULL` on the first call, meaning we’d need to handle this special case with an extra `if` statement. Like this:

``` c
static char *
eager_strpbrk(const char *string, const char *separators) {
	char *ptr, *tmp_ptr;
	ptr = tmp_ptr = strpbrk(string, separators);

	if (ptr == NULL) {
		return NULL;
	}

	while ((tmp_ptr = strpbrk(++tmp_ptr, separators))) {
		ptr = tmp_ptr;
	}
	
	return ptr;
}
```

On the contrary, `for` loop handles NULL pointers naturally. But if you still think that `while`-based implementation is clearer or more consistent with the rest of the project’s codebase, I don’t mind changing my commit to it.